job_definitions/update_api_tokens.yml

```diff
@@ -0,0 +1,74 @@
+{% set stages = [{'name': 'Add new tokens', 'command': 'add-new'}, {'name': 'Remove old tokens', 'command': 'remove-old'}] %}
+{% set app_groups = [['search-api'], ['api'], ['admin-frontend', 'briefs-frontend', 'brief-responses-frontend', 'buyer-frontend', 'supplier-frontend', 'user-frontend']] %}
+---
+- job:
+    name: update-api-tokens
+    display-name: "Update API tokens"
+    project-type: pipeline
+    descript1ion: Update Data and Search API tokens for a given environment.
+    disabled: false
+    concurrent: false
+    parameters:
+      - choice:
+          name: STAGE
+          choices:
+            - preview
+            - staging
+            - production
+    pipeline:
+      script: |
+        def notify_slack(icon, status) {
+          build job: "notify-slack",
+          parameters: [
+            string(name: 'USERNAME', value: "update_api_tokens"),
+            string(name: 'ICON', value: icon),
+            string(name: 'JOB', value: "Update API tokens on ${STAGE}"),
+            string(name: 'CHANNEL', value: "#dm-release"),
+            text(name: 'STAGE', value: "${STAGE}"),
+            text(name: 'STATUS', value: status),
+            text(name: 'URL', value: "<${BUILD_URL}consoleFull|${BUILD_DISPLAY_NAME}>")
+          ]
+        }
+
+{% for stage in stages %}
+        stage("{{ stage.name }}") {
+{% if loop.index == 1 %}
```

Comment:
This threw me a bit, why you were using index '1'. Had to look at the docs to find out `loop.index` is 1 indexed. `loop.index()` is 0 indexed. Would it be more intuitive to use `loop.index()`

Reply:
There's `index` and `index0` I think. Haven't heard of `index()` but if that also exists it's super weird.

Reply:
derp. I read `0` as`()`